DP/SimpleCongruenceClosure.hpp

```diff
@@ -100,11 +100,11 @@ class SimpleCongruenceClosure : public DecisionProcedure
   /** Equality between two constants */
```

Comment:
Some constructors do not initialise some members - not a problem _if everything is initialised before use_, but let's make it deterministic and cause a null-pointer exception if members are not subsequently initialised later.

Reply:
All of the constructor fixes here look sensible.